### PR TITLE
Overlay status icons over base icon

### PR DESCRIPTION
**Pull request by fmoc**
This way, we do not need to provide pre-rendered icons to depict the different synchronization states but can dynamically overlay small icons over the base application icon.

As a fallback, the existing setup will continue to be supported: as long as an icon called "base" is not found, the lookup assumes the overlay icon filename contains a pre-rendered icon and therefore returns it unmodified.

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-oc10</strong> https://drone.owncloud.com/owncloud/client/14664/6/1

<details>


</details>

**Comment by TheOneRing** [comment]
Please fix your commits before review...

**Comment by fmoc** [comment]
This branch can be squash-merged just fine. I can also squash it upon approval in a matter of seconds. I'm not sure what advantage you see in squashing in advance of a review.

**Review by TheOneRing — CHANGES_REQUESTED** [review]
Code is not ready for review

File: src/libsync/overlayiconengine.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+OverlayIconEngine::OverlayIconEngine(const QIcon &base, const QIcon &overlay)
+    : _base(base)
+    , _overlay(overlay)
+{
+}
+
+void OverlayIconEngine::paint(QPainter *painter, const QRect &rect, QIcon::Mode mode, QIcon::State state)
+{
+    // TODO: copied from qsvgiconengine
```
Comment (TheOneRing) [inline src/libsync/overlayiconengine.cpp:13]:
```suggestion
    // Based on qsvgiconengine, compute a dpi aware scaled size
```

File: src/libsync/theme.cpp [hunk 2]
```diff
@@ -469,64 +470,76 @@ bool Theme::aboutShowCopyright() const
...
-    const auto statusIcon = QLatin1String("state-offline");
-    if (sysTray) {
-        return themeTrayIcon(statusIcon, sysTrayMenuVisible);
+        return QIcon(new OverlayIconEngine(baseIcon, overlayIcon));
     } else {
-        return themeIcon(statusIcon);
+        auto baseIcon = themeIcon(QStringLiteral("base"), Theme::IconType::BrandedIcon);
+        auto overlayIcon = themeIcon(statusIconName);
+
+        if (baseIcon.isNull()) {
```
Comment (TheOneRing) [inline src/libsync/theme.cpp:538]: We could unify the code paths minimally.

File: src/resources/core_theme.qrc [hunk 3]
```diff
@@ -6,26 +6,33 @@
...
+        <file alias="theme/dark/base.svg">theme/dark/ui-dark-plain-color-base.svg</file>
+
         <file alias="theme/white/state-ok.svg">theme/white/ui-dark-plain-monochrom-state-checkmark.svg</file>
         <file alias="theme/white/state-error.svg">theme/white/ui-dark-plain-monochrom-state-error.svg</file>
         <file alias="theme/white/state-information.svg">theme/white/ui-dark-plain-monochrom-state-info.svg</file>
         <file alias="theme/white/state-offline.svg">theme/white/ui-dark-plain-monochrom-state-offline.svg</file>
         <file alias="theme/white/state-pause.svg">theme/white/ui-dark-plain-monochrom-state-pause.svg</file>
         <file alias="theme/white/state-sync.svg">theme/white/ui-dark-plain-monochrom-state-sync.svg</file>
+        <file alias="theme/white/base.svg">theme/white/ui-dark-plain-monochrom-base.svg</file>
+
```
Comment (TheOneRing) [inline src/resources/core_theme.qrc:33]: The file was formated by qt creator, not sure manual formatting is a good idea.

File: src/resources/theme/colored/sidebar.svg
Comment (TheOneRing) [inline src/resources/theme/colored/sidebar.svg:1]: its either the preview or we had a color change, why was that icon touched at all?

**Review by TheOneRing — CHANGES_REQUESTED** [review]

File: src/libsync/overlayiconengine.cpp [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    if (painter->device())
+        pixmapSize *= painter->device()->devicePixelRatio();
+    painter->drawPixmap(rect, pixmap(pixmapSize, mode, state));
+}
+
+QPixmap OverlayIconEngine::pixmap(const QSize &size, QIcon::Mode mode, QIcon::State state)
+{
+    QImage img(size, QImage::Format_RGBA64_Premultiplied);
+    // QImage's pixels are not initialized, therefore first things first: fill image with transparency
+    img.fill(0);
```
Comment (TheOneRing) [inline src/libsync/overlayiconengine.cpp:24]: Qt::transparent

**Comment by TheOneRing** [comment]
Lets discuss next week

**Pull request closed without merging**